R/layouts.R

```diff
@@ -0,0 +1,164 @@
+# Layouts
+
+#' Creates row wrapping specified elements
+#' @param arg Element to be wrapped in row container
+#' @return character with row wrapping specified elements
+get_row <- function(arg) {
+  class <- "row"
+  style <- "padding: 20px;"
+  shiny::HTML(glue::glue("<div class='{class}' style='{style}'>{arg}</div>"))
+}
+
+#' Creates div containing children elements in separate rows
+#'
+#' @param ... Container's children elements
+#' @return Div containing children elements in separate rows
+panel <- function(...) {
+  args <- list(...)
+  div(lapply(args, get_row))
+}
+
+#' Creates div containing children elements of sidebar panel
+#'
+#' @param ... Container's children elements
+#' @param width Width of sidebar panel container as relative value
+#' @rdname sidebar_layout
+#' @export
+sidebar_panel <- function(..., width = 1) {
+  list(panel = panel(...), width = width)
+}
+
+#' Creates div containing children elements of main panel
+#'
+#' @param ... Container's children elements
+#' @param width Width of main panel container as relative value
+#' @rdname sidebar_layout
+#' @export
+main_panel <- function(..., width = 3) {
+  list(panel = panel(...), width = width)
```

Comment:
Why does `main_panel` have to render its arguments in rows like `sidebar_panel`? It doesn't work like this in Shiny's `mainPanel`: https://shiny.rstudio.com/reference/shiny/0.11/mainPanel.html (I couldn't find reference for Shiny 1.5.0)

I thought I can put any HTML elements here and organise them by myself.

Reply:
I changed it to:
```
main_panel <- function(..., width = 3) {
  list(children = list(...), width = get_numeric(width))
}
```
but maybe we could write it in more optimized way?